modules/shared-vpc-helper/main.tf

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+locals {
+  projects               = [var.gke_svpc_host_project, var.gke_svpc_service_project]
+  service_project_number = element(coalescelist(data.google_project.service_project[*].number, ["null"]), 0)
+
+  gke_s_account           = "serviceAccount:service-${local.service_project_number}@container-engine-robot.iam.gserviceaccount.com"
+  gke_default_s_account   = "serviceAccount:${local.service_project_number}@cloudservices.gserviceaccount.com"
+  shared_vpc_users_length = 3
+  shared_vpc_users        = [local.gke_s_account, local.gke_default_s_account, var.gke_sa]
+}
+
+/******************************************
+	Container API enable
+ *****************************************/
+
+resource "google_project_service" "gke_api" {
```

Comment:
This should make use of the [project services submodule](https://github.com/terraform-google-modules/terraform-google-project-factory/tree/master/modules/project_services/).